test/e2e/network/endpointslice.go

```diff
@@ -145,7 +145,7 @@ var _ = SIGDescribe("EndpointSlice", func() {
 		}
 
 		// Expect EndpointSlice resource to be deleted when Service is.
-		if err := wait.PollImmediate(2*time.Second, wait.ForeverTestTimeout, func() (bool, error) {
```

Comment:
a comment here would be good as to why this was selected

Reply:
I don't have a great reason other than to deflake tests. I'm not sure what a reasonable amount of time to wait for garbage collection to complete is, does this feel like a reasonable timeout?

Reply:
GC discovers served resource types every 30 seconds, and if they have changed (which they do frequently in e2e suites creating/deleting CRDs) it pauses GC while it sets up informers for the new types. If one of the discovered resource types becomes unavailable, the informer will not sync, and GC will timeout waiting for it after 30 seconds and rediscover available resources. I would add 1 minute for allowed GC controller-pausing overhead on top of whatever timeout you would wait for the resource itself to be cleaned up.

Reply:
https://github.com/kubernetes/kubernetes/blob/f64f687fcfb22e5099fbcb990f3e6faa44446fff/test/e2e/apimachinery/garbage_collector.go#L102-L110

Reply:
Thanks! That's very helpful - just updated the comment and increased the timeout to 90 seconds.